### PR TITLE
Added ScanServerShutdownIT to confirm refs deleted on shutdown

### LINKED ISSUE #4594: Add IT to verify Scan Servers remove their references on shutdown
**Is your feature request related to a problem? Please describe.**
Noticed some behavior where scan servers shutdown via a `./accumulo-cluster stop-servers` command would not clean up their entries from the metadata table.

**Describe the solution you'd like**
Add an IT test to verify that the scan servers delete their references on a expected shutdown pattern.

### PR DESCRIPTION
**Pull request by dlmarion**
Fixes #4594

### CODE REVIEW

File: test/src/main/java/org/apache/accumulo/test/ScanServerShutdownIT.java [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+
+      client.tableOperations().create(tableName);
+
+      // Make multiple files
+      final int fileCount = 3;
+      for (int i = 0; i < fileCount; i++) {
+        ScanServerIT.ingest(client, tableName, 10, 10, 0, "colf", true);
+      }
+
+      for (int i = 0; i < 3; i++) {
```
Comment (keith-turner) [inline test/src/main/java/org/apache/accumulo/test/ScanServerShutdownIT.java:111]: On the 2nd iteration could verify that some refs exists.

```suggestion
      for (int i = 0; i < 3; i++) {
         if(i ==1){
             // the 1st scan should have created some refs
             assertTrue(ctx.getAmple().getScanServerFileReferences().count()>0);
         }
```

File: test/src/main/java/org/apache/accumulo/test/ScanServerShutdownIT.java [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+      final String tableName = getUniqueNames(1)[0];
+
+      client.tableOperations().create(tableName);
+
+      // Make multiple files
+      final int fileCount = 3;
+      for (int i = 0; i < fileCount; i++) {
+        ScanServerIT.ingest(client, tableName, 10, 10, 0, "colf", true);
+      }
+
```
Comment (keith-turner) [inline test/src/main/java/org/apache/accumulo/test/ScanServerShutdownIT.java:110]: Can verify there are no refs here.

```suggestion
assertEquals(0, ctx.getAmple().getScanServerFileReferences().count());
```

**Review by keith-turner — APPROVED** [review]

File: test/src/main/java/org/apache/accumulo/test/ScanServerShutdownIT.java (hunk 1 above)
Comment (dlmarion) [reply test/src/main/java/org/apache/accumulo/test/ScanServerShutdownIT.java:111]: Do they get cleaned up when the scan closes? If not, then the check on line 120 below should fail, right? It should be 3, 6, 9 and not 3 every time?

File: test/src/main/java/org/apache/accumulo/test/ScanServerShutdownIT.java (hunk 2 above)
Comment (dlmarion) [reply test/src/main/java/org/apache/accumulo/test/ScanServerShutdownIT.java:111]: Added in 89797f3

File: test/src/main/java/org/apache/accumulo/test/ScanServerShutdownIT.java (hunk 1 above)
Comment (keith-turner) [reply test/src/main/java/org/apache/accumulo/test/ScanServerShutdownIT.java:112]: Yeah the check on line 120 is sufficient, I missed that.   I think I glossed over that code block assuming it was just scanning.

**Review by keith-turner — APPROVED** [review]

**Pull request merged**